### PR TITLE
changed professional codes to sds in the example messages

### PR DESCRIPTION
**Pull request by mukhtar-abdul-ghani**

### CODE REVIEW

File: examples/secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request/1-Convert-Response-Claim-200_OK.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
                 <code code="R8000" codeSystem="1.2.826.0.1285.0.2.1.104"/>
                 <telecom use="WP" value="tel:0532567890"/>
                 <agentPerson classCode="PSN" determinerCode="INSTANCE">
-                  <id extension="7654321" root="1.2.826.0.1285.0.2.1.54"/>
+                  <id extension="3415870201" root="1.2.826.0.1285.0.2.0.65"/>
```
Comment (dan-jones-iw) [inline examples/secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request/1-Convert-Response-Claim-200_OK.xml:74]: bit worried about this one - i THINK the requirements ask us to populate OID ending with `1.54`, just the value should be an SDS ID. think we should hold off on this until we get this clarified - probs by ameya

**Review by dan-jones-iw — COMMENTED** [review]
:+1:

File: examples/secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request/1-Convert-Response-Claim-200_OK.xml (hunk 1 above)
Comment (TomSmithCoding) [reply examples/secondary-care/community/repeat-dispensing/nominated-pharmacy/clinical-practitioner/single-medication-request/1-Convert-Response-Claim-200_OK.xml:74]: Confirmed with Ameya (who confirmed with Rob Gooch) it does infact need to be the professional code root (1.54) with the SDS ID value.

File: coordinator/src/services/translation/request/agent-unattended.ts [hunk 2]
```diff
@@ -41,12 +40,17 @@ export function createAuthorForWithdraw(
       'Task.contained("PractitionerRole").practitioner("value")'
     )
   }
-  const sdsUserUniqueId = getAgentPersonPersonIdForAuthor([practitionerRole.practitioner.identifier])
+
+  const sdsUserUniqueId = getIdentifierValueForSystem(
+    [practitionerRole.practitioner.identifier],
+    "https://fhir.nhs.uk/Id/sds-user-id",
+    'Task.contained("PractitionerRole").practitioner("value")'
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/request/agent-unattended.ts:47]: this isnt going to always be a task is it?

File: coordinator/tests/services/translation/request/agent-unattended.spec.ts [hunk 3]
```diff
@@ -112,7 +102,7 @@ describe("createAuthorForWithdraw", () => {
       .extension
     )
       .toBe(
-        professionalCodeExpected
+        sdsCodeExpected
```
Comment (dan-jones-iw) [inline coordinator/tests/services/translation/request/agent-unattended.spec.ts:105]: can we be consistent in how we do these expectations:
line 41 - `expect(id).toStrictEqual(new SDSUniqueId(value))`
line 105 - `expect(id.attributes.extension).toBe(value)`

File: coordinator/src/services/translation/request/agent-unattended.ts [hunk 4]
```diff
@@ -41,12 +39,17 @@ export function createAuthorForWithdraw(
...
+  )
 
   const agentPersonSds = new hl7V3.AgentPersonSds()
   agentPersonSds.id = new hl7V3.SdsRoleProfileIdentifier(sdsRoleProfileId)
   agentPersonSds.agentPersonSDS = new hl7V3.AgentPersonPersonSds(
-    new hl7V3.SdsUniqueIdentifier(sdsUserUniqueId._attributes.extension)
+    new hl7V3.ProfessionalCode(sdsUserUniqueId)
```
Comment (dan-jones-iw) [inline coordinator/src/services/translation/request/agent-unattended.ts:52]: `//we want OID ending in 1.54 because of decision 010'

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**